src/Dfe.PlanTech.Web/ViewComponents/CategorySectionViewComponent.cs

```diff
@@ -53,13 +60,17 @@ private void LogErrorWithUserFeedback(Section categorySection, CategorySectionDt
 
     private static void SetCategorySectionDtoTagWithRetrievalError(CategorySectionDto categorySectionDto)
     {
-        categorySectionDto.TagColour = TagColour.Red.ToString();
+        categorySectionDto.TagColour = TagColour.Red;
         categorySectionDto.TagText = "UNABLE TO RETRIEVE STATUS";
     }
 
-    private static void SetCategorySectionDtoTagWithCurrentStatus(Category category, Section categorySection, CategorySectionDto categorySectionDto)
+    private static void SetCategorySectionDtoTagWithCurrentStatus(
+        Category category,
+        Section categorySection,
+        CategorySectionDto categorySectionDto)
     {
-        var sectionStatusCompleted = category.SectionStatuses.FirstOrDefault(sectionStatus => sectionStatus.SectionId == categorySection.Sys.Id)?.Completed;
+        var sectionStatusCompleted = category.SectionStatuses
```

Comment:
My initial thought here was
"We've already got the section status for the section on line 99 - do we need this?"

but then I realise on line 99 that you only get the _completed_ section status at that point.

Based on that however, my thought now is:
Given the SQL stored procedure you created, will we ever have more than one section statuses for a section? I.e. don't we only return the latest for each section? If a user has previously completed a section, but it is now in progress again, don't we want both statuses?

Reply:
I dunno what the existing functionality is FWIW. I assume we also already only use the latest status for each section.

if that's the case though, you could probably refactor to remove this LINQ statement.

Reply:
Good shout - we're currently using the latest status for each, but you're right, on line 99 we get the completed status, but we can just get the status instead and check for completeness when looking for the recommendation